pyasn1/codec/ber/decoder.py

```diff
@@ -67,16 +106,14 @@ def valueDecoder(self, substrate, asn1Spec,
                 self._createComponent(asn1Spec, tagSet, '', **options),
                 substrate, length
             )
+        value = decodeFun(substrate, asn1Spec, tagSet, length, **options)
 
-        head, tail = substrate[:length], substrate[length:]
-
-        value, _ = decodeFun(head, asn1Spec, tagSet, length, **options)
-
-        if LOG:
-            LOG('explicit tag container carries %d octets of trailing payload '
-                '(will be lost!): %s' % (len(_), debug.hexdump(_)))
+        # TODO: Don't know what to with this
```

Comment:
I think the original code misses the `if _:` condition.

I am guessing there should still be a place to cut the decoding by `length` as specified in the header (TLV)...? Then what's left beyond the `length` are those extra octets.

Reply:
Sure. It's not that hard to solve the "Don't know" ;-)